Fusion reactions

biscuitWizard opened the pull request:
Added new fusion reactions + some new meta, as well as new engine propellant (radium, 2x heavier than CO2, per realism.)

biscuitWizard committed:
Merge pull request #1 from Baystation12/dev

Merging down from upstream

Commit:
Tweaked boron and lithium reactions for balance.

Commit:
Adding radium gas.

Commit:
Added a uranium->radium reaction

Commit:
Radium when breathed now turns into reagent radium in the body.

Commit:
Adding uranium fuel assembly.

Commit:
Added graphite-graphite reaction.
Stabilized some other reactions for balance reasons.
Replaced many strings with defines.

Commit:
Added change log.

biscuitWizard commented:
Another commit will be pulled in shortly. I've refactored fusion_plant's network system into a generic local_network pattern so it can be reused elsewhere. Est. that will happen tonight.

PsiOmegaDelta commented:
We have a generic network system via https://github.com/Baystation12/Baystation12/tree/dev/code/datums/graph

biscuitWizard commented:
Hmm,  I don't think that fits the same functionality as a local network.

Commit:
Adding local network abstraction of the fusion plant network, which will also be used for airlocks and the BSD rework.

Local networks are groups of proximity-bound devices set by their network identifier. In the future this should likely be replaced with radio-component based communication, but for simplicity and surgical replacement, this abstraction is provided as just a slightly modified fusion_plant network.

biscuitWizard commented:
Added the network abstraction. No changelog required as it's not a player-facing change. Tested and fusion still works.

biscuitWizard commented:
Closing for now, I'm going to do some additional work on this branch. Will reopen later.
